File: java-app/src/main/java/com/prosoft/java/controller/Controller.java
package com.prosoft.java.controller;

import com.prosoft.java.service.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/v1")
public class Controller {

    private final UserService userService;

    @Autowired
    public Controller(UserService userService) {
        this.userService = userService;
    }

    @GetMapping("/hello")
    public String greeting() {
        return "Hello, Java!";
    }

    @GetMapping(value = "/user", params = "id")
    public String greetingUser(@RequestParam(name = "id") Long id) {
        return String.format("Hello, %s!", userService.getUser(id));
    }

}

File: java-app/src/main/kotlin/com/prosoft/kotlin/controller/Controller.kt
package com.prosoft.kotlin.controller

import com.prosoft.kotlin.service.UserService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController("kotlinController")
@RequestMapping("/v2")
class Controller @Autowired constructor(private val userService: UserService) {

    @GetMapping("/hello")
    fun greeting(): String {
        return "Hello, Kotlin!"
    }

    @GetMapping(value = ["/user"], params = ["id"])
    fun greetingUser(@RequestParam(name = "id") id: Long):String {
        return "Hello, ${userService.getUser(id)}!"
    }

}